cstar/base/runtime_env_config.py

```diff
@@ -0,0 +1,136 @@
+import platform
+import os
+import importlib.util
+from pathlib import Path
+from contextlib import redirect_stderr, redirect_stdout
+import io
+from cstar.base.hpc_env_var_map import determineHPCEnvVars
+
+class RuntimeEnvConfig:
+
+    def __init__(self, _CSTAR_ROOT):
+        self.envVars = {}
+        
+        self.system = platform.system()
+        self.hasLMODDir = "LMOD_DIR" in list(os.environ)
+        
+        self._CSTAR_ROOT = _CSTAR_ROOT
+        
+        self._CSTAR_COMPILER = ''
+        self._CSTAR_SYSTEM = ''
+        self._CSTAR_SCHEDULER = ''
+        self._CSTAR_ENVIRONMENT_VARIABLES = ''
+        self._CSTAR_SYSTEM_DEFAULT_PARTITION = ''
+        self._CSTAR_SYSTEM_CORES_PER_NODE = ''
+        self._CSTAR_SYSTEM_MEMGB_PER_NODE = ''
+        self._CSTAR_SYSTEM_MAX_WALLTIME = ''
+        
+        if (self.system == "Linux") and ("LMOD_DIR" in list(os.environ)):
+            self.configureHPCRuntimeEnv()
+        else:
+            self.configureLocalEnv() 
+    
+    
+    def loadEnvModule():
```

Comment:
`LoadEnvModule()` and `loadLinuxEnvModule()` are broken out from the original if statement in `environment.py`

Perhaps the names could be revised to be more contextually meaningful.